frontend/src/components/transactions/TransactionDetails.tsx

```diff
@@ -36,11 +40,31 @@ export default class extends React.Component<Props, State> {
       })
       .reduce((accumulator, deposit) => accumulator.add(deposit), new BN(0));
     this.setState({ deposit });
+  };
+
+  setTotalFee = () => {
+    const price = new BN(this.props.transaction.gasPrice);
+    const txFee = this.props.transaction.transactionOutcome
+      ? new BN(this.props.transaction.transactionOutcome.outcome.gas_burnt)
+      : new BN(0);
+    const receiptFee = this.props.transaction.receiptsOutcome
+      ? this.props.transaction.receiptsOutcome
+          .map(receipt => new BN(receipt.outcome.gas_burnt))
+          .reduce((gasBurnt, currentFee) => gasBurnt.add(currentFee), new BN(0))
+      : new BN(0);
+    const gasUsed = txFee.add(receiptFee);
+    const totalFee = gasUsed.mul(price);
+    this.setState({ totalFee, gasUsed });
```

Comment:
@bowenwang1996 @ilblackdragon @nearmax I have seen a conversation somewhere mentioning that it is quite hard to compute the fee as you seem to need to collect the gas price from the relevant block where the receipt was issued (which is not the transaction where it gets included). Am I recalling this right? It seems that we cannot match the receipt with the transaction as RPC does not expose any link between them (e.g. receipt does not have a transaction hash or anything like that). :disappointed:

Reply:
what I understand is you have said we should "sum" them (transaction outcome and receipt outcome) let's best talk with runtime people to make sure what it is

Reply:
My understanding is that we should sum (pseudo-code) `receipt.gas_burnt * receipt.transaction.block.gas_price` (where `receipt.block` is not the same as `receipt.transaction.block`, so we may end up querying `gas_price` from several blocks to compute the transaction fee in NEAR)

Reply:
@frol it should be the sum of (transaction gas burnt + all receipt gas burnt) * gas price

Reply:
To collect the fee in gas you need to sum all gas usage across all receipts issued by the transaction, including the transaction itself. To convert this fee from gas to tokens you only need to know the gas price of the block that includes the transaction, this price is included in every receipt produced by the transaction: https://github.com/nearprotocol/nearcore/blob/4713f5f88a7a57d6b5a13ed1057289858d63f134/core/primitives/src/receipt.rs#L76

Reply:
It seems that this computation is implemented correctly, but I would fix the variable names:

* `price` -> `gasPrice`
* `txFee` -> `gasBurntByTx`
* `receiptFee` -> `gasBurntByReceipts`
* `totalFee` -> `fee` (or `transactionFee`, but then we also should align `gasUsed` by renaming it to `transactionGasBurnt`, which sounds ridiculous)

Also, I think these computations (deposit as well) are generic enough to move them right into the TransactionApi layer. Thus, you will receive all the computed values "from API".

Reply:
`gasAttached` - that either `==` to `gasUsed` for non `FunctionCall` actions or equal to attached gas in the `FunctionCall` Action.